scale-encode/src/lib.rs

```diff
@@ -183,50 +178,59 @@ pub trait EncodeAsType {
 /// tuple and struct types, and is automatically implemented via the [`macro@EncodeAsType`] macro.
 pub trait EncodeAsFields {
     /// Given some fields describing the shape of a type, attempt to encode to that shape.
-    fn encode_as_fields_to(
+    fn encode_as_fields_to<'a, I: FieldIter<'a>>(
```

Comment:
Adding this generic parameter makes it no longer object safe so breaks https://github.com/paritytech/subxt/blob/ae0ab3e8f332050311e3eccdc156507e0c06070a/subxt/src/tx/tx_payload.rs#L99